unimod_mapper/unimod_mapper.py

```diff
@@ -498,6 +517,64 @@ def _map_key_2_index_2_value(self, map_key, return_key):
     def _data_list_2_value(self, index, return_key):
         return self.data_list[index][return_key]
 
+    def writeXML(self, modification_dict, xml_file=None):
```

Comment:
help me to clear my mind...

In principle we will have the `unimod.xml` and `userdefined_unimod.xml` files. The first one will be fetched from unimod directly and NOT touched. the latter one will be appended with user defined modifications. I assume the only reason for having `xml_file` as an arg is for the test purpose. However, if somebody comes up with the nice idea of calling his userdefined_unimod file differently (e.g. usermods.xml) the `_reparseXML `will not work as expected as the file will not be considered as you execute `self._parseXML()`. You can see it in the test, since you have to execute the _parseXML again outside of the writeXML function to make the right assert. And while you do so, you actually only parse the `test_only_unimod.xml` and not the additionally original unimod.xml anymore. 

I am just wondering if we should have this option that users can define the file name by themselves in the first place... If our idea how the original and user defined files should co-exist is the one I pointed out, the current code will not work as expected once the user defines a name for `xml_file` other than `userdefined_unimod.xml`

Or do we want to have an option for multiple `userdefined_unimod.xml` files (e.g. for different projects etc.)? If so, we might consider appending the `xml_file` to the `self.unimod_xml_names` and refactor the parser.

Reply:
Having the ability to define input xml files is not only for test purposes but also for when people rename their usedefined xml files, have different ones, etc.
But you're right, it wasn't well implemented, I basically just took the functionality of the old version, but have added the option to pass down an xml file list now. The files passed down that way will be included in addition to the original unimod.xml and the default userdefined_unimod.xml (if it exists).